packages/flutter_tools/lib/src/asset.dart

```diff
@@ -457,17 +466,17 @@ class ManifestAssetBundle implements AssetBundle {
   @override
   List<File> additionalDependencies = <File>[];
 
-  void _setIfChanged(String key, DevFSStringContent content, AssetKind assetKind) {
-    if (!entries.containsKey(key)) {
-      entries[key] = content;
-      entryKinds[key] = assetKind;
-      return;
-    }
-    final DevFSStringContent? oldContent = entries[key] as DevFSStringContent?;
-    if (oldContent?.string != content.string) {
-      entries[key] = content;
-      entryKinds[key] = assetKind;
-    }
+  void _setIfChanged(String key, DevFSContent content, AssetKind assetKind) {
+    final DevFSContent? oldContent = entries[key];
+
+    // TODO(andrewkolos): Why is the "IfChanged" part important?
```

Comment:
if the object is not backed by a file or something, then the tool will assume that new entiries need to be updated and these will get copied to the device on each hot reload/restart, which is wasteful

Reply:
Where/when would this assumption happen? Is there some caching mechanism that compares values by reference?